src/mount.js

```diff
@@ -64,14 +87,22 @@ class MountDatastore extends Adapter {
     return match.datastore.get(match.rest, options)
   }
 
+  /**
+   * @param {Key} key
+   * @param {Options} [options]
+   */
   has (key, options) {
     const match = this._lookup(key)
     if (match == null) {
-      return false
+      return Promise.resolve(false)
```

Comment:
Return types are inferred from the interface. However, it would probably be helpful to have them in the code, special for returning promises. What do you think?

Reply:
@Gozala for that we would need to make it an async function
@vasco-santos i would like to avoid writing all the return types if they are already well inferred

Reply:
> @Gozala for that we would need to make it an async function

I don't believe that's accurate, in fact that was the whole point of `Await` type. [Here is an example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAgg7gQwJbADwBUB8UC8UAKATgPYC2SAzhBtgD5ToBQjSAdsBIQGYIDG0AaQggoAb0ZRJUYMQDKwQmwDmACgCUALgoLljAL7M2Hbn2gAJBBTESpAC0sqA1sI1CQAGijEwwJMVYUAPwaohRISqwIADbBsABGxITAsuGRUXqa8MhoCcRREAismPrMvFGWVvKJ0EikYPmkEOxWFlbiUlD2FE4ubp7evv4UIWER0bEwCUkpY+mZiCioufmF2O0dUkhcUD0gAHQy8oqsqmp72ghJFADqKLYqAOS2wMBgGgD07w9qatYbG4QIMAAK6EVhQLhA3j3Zz7Q46E7qM7AWxNFSAihgIbQHDYDFYgIQPbERxqGz-PRQCBRKh-f4AoGg8E8GkQckbAwdAx6IA):

```ts
type Await<T> = Promise<T> | T

interface Key {
    toString():string
}

interface Has {
    has(key:Key, options?:{signal?: AbortSignal}):Await<boolean>
}

class Store implements Has {
    has(key:Key, options:{signal?: AbortSignal}):Await<boolean> {
        if (key.toString().startsWith('http://')) {
          return fetch(key.toString()).then(response => response.ok)
        } else {
            return false
        }
    }
}
```

Reply:
As a pattern I think `Await<T>` is something we should allow.  The idea of the return type being variable is a bit weird, but in the past we've desired the ability to not create artificial asynchronicity for super-hot codepaths, the idea being you'd look at the return type and resolve the promise if it's a thenable.

Though it's not something we should be doing everywhere, blindly `await`ing will be fine 99.9% of the time, unless some rigorous profiling has identified a bottleneck.

Reply:
It does make the error handling for the caller a bit ~Zalgo~ harder to handle, mind:

```js
async function derp (input) {
  if (input % 2 === 1) {
    throw new Error('Input was odd')
  }

  return Promise.reject(new Error('Input was even'))
}
```

```js
const res1 = derp(1) // throws immediately

const res2 = derp(2)

// maybe do some other stuff

await res2 // throws here instead
```

Reply:
what do you think we should do here ?